apps/desktop/src/app/tools/send/new-send/new-send-dropdown.component.ts

```
@@ -0,0 +1,61 @@
+import { CommonModule } from "@angular/common";
+import { Component, EventEmitter, Input, Output } from "@angular/core";
+import { firstValueFrom, Observable, of, switchMap } from "rxjs";
+
+import { JslibModule } from "@bitwarden/angular/jslib.module";
+import { AccountService } from "@bitwarden/common/auth/abstractions/account.service";
+import { BillingAccountProfileStateService } from "@bitwarden/common/billing/abstractions";
+import { MessagingService } from "@bitwarden/common/platform/abstractions/messaging.service";
+import { SendType } from "@bitwarden/common/tools/send/enums/send-type";
+import { BadgeModule, ButtonModule, MenuModule } from "@bitwarden/components";
+
+@Component({
+  selector: "tools-new-send-dropdown",
+  templateUrl: "new-send-dropdown.component.html",
+  standalone: true,
+  imports: [JslibModule, CommonModule, ButtonModule, MenuModule, BadgeModule],
+})
+/**
+ * A dropdown component that allows the user to create a new Send of a specific type.
+ */
+export class NewSendDropdownComponent {
+  /** If true, the plus icon will be hidden */
+  @Input() hideIcon: boolean = false;
+
+  /** SendType provided for the markup to pass back the selected type of Send */
+  protected sendType = SendType;
+
+  /** Indicates whether the user can access premium features. */
+  protected canAccessPremium$: Observable<boolean>;
+
+  /** Emitted when an allowed SendType has been selected. */
+  @Output() onCreateSendOfType = new EventEmitter<SendType>();
+
+  constructor(
+    private billingAccountProfileStateService: BillingAccountProfileStateService,
+    private accountService: AccountService,
+    private messagingService: MessagingService,
+  ) {
+    this.canAccessPremium$ = this.accountService.activeAccount$.pipe(
+      switchMap((account) =>
+        account
+          ? this.billingAccountProfileStateService.hasPremiumFromAnySource$(account.id)
+          : of(false),
+      ),
+    );
```

🤔 The situation where there's no active account seems like it should not ever occur. If there are states where an account could be null, they're anomalous and fall into one of two categories:

* A transition state that occurs during account switching and/or locking that is safe to ignore, OR
* A state that should never occur.

In the former case, you might consider hoisting the ternary from the `switchMap` to a `filter((a) => !!a)`. In the latter you might consider throwing instead of continuing with a disabled value.

💭 This could also indicate that the component is too smart. If the account should always be required, consider [requiring the component consumer to supply it instead](https://github.com/bitwarden/clients/blob/main/libs/tools/generator/components/src/passphrase-settings.component.ts#L49-L60).